geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/ArrowAttributeReader.scala

```diff
@@ -150,22 +141,65 @@ object ArrowAttributeReader {
         }
 
       case Some(dict) =>
-        val dictionaryType = TypeBindings(bindings, encoding)
         vector match {
-          case v: TinyIntVector  => new ArrowDictionaryByteReader(v, dict, dictionaryType)
-          case v: SmallIntVector => new ArrowDictionaryShortReader(v, dict, dictionaryType)
-          case v: IntVector      => new ArrowDictionaryIntReader(v, dict, dictionaryType)
+          case v: TinyIntVector  => new ArrowDictionaryByteReader(v, dict)
+          case v: SmallIntVector => new ArrowDictionaryShortReader(v, dict)
+          case v: IntVector      => new ArrowDictionaryIntReader(v, dict)
+
+          case v: ListVector =>
```

Comment:
Going back to my above question.  Is this covering the cases where one could have List[TinyInt/SmallInt] which are separate from Dictionary encoded situations?

Reply:
right - if you know the number of dictionary entries up front, you can create a smaller encoding space. for the delta writer, since we don't know the sizes up front, we just encode everything as 32-bit ints. so the logic in that class doesn't have to deal with tiny/small ints.